License server setup docs (#421)
metalbear-co/mirrord.dev: content/docs/managing-mirrord/license-server/index.md

```bash
# will launch the operator
docker run ghcr.io/metalbear-co/operator:3.107.2
```

```bash
# will launch the license-server
docker run ghcr.io/metalbear-co/operator:3.107.2 license-server
``` 

Though they do use some distinct configuration args and env, and thus we recommend using the helm chart.

### Helm Chart

Simplest deployment is with chart `mirrord-license-server`
```yaml
# ./values.yaml
createNamespace: true

service:
  type: ClusterIP

license:
  key: secret
  file:
    data:
      license.pem: |
        ----- ... 
        MIRRORD-LICENSE 
        ... -----
```
*NOTE: default value for `service.type` is `ClusterIP`, this service should be accesible to the operator so you might concider using `NodePort` or `LoadBalancer` types*

Keep in mind that the `license.key` can be any value and should be used as `--license-key` in operator setup.

<br />

With this `values.yaml` we can install the license server on a kubernetes cluster.

```bash
helm repo add metalbear-co https://metalbear-co.github.io/charts
 helm install metalbear-co/mirrord-operator-license-server -f ./values.yaml --generate-name
```

Once it's installed and running now you can connect the operator.

> Depending if installed the operator via helm or cli you can specify the following properties.

Reviewer: Lets start a new section here - Connecting Operators to the License Server

```bash
mirrord operator setup --accept-tos --license-key secret --license-server http://<license-server-addr> | kubectl apply -f -
```

or if using with Helm chart
```yaml
# ./values.yaml
license:
  key: secret
  licenseServer: http://<license-server-addr>
```
and then use the install or upgrade to set these parameters
```bash
helm install metalbear-co/mirrord-operator -f ./values.yaml --generate-name
```

The `<license-server-addr>` should be an acceible endpoint of the `mirrord-operator-license-server` service meaning if you have it installed under `LoadBalancer` confuguration it will be the relevant `<external-ip>:<node-port>`

> If you have an ingress installed on the cluster please do expose the `mirrord-operator-license-server` service via http or https to license server http/https endpoint (depends on `tls` and `service.port` values in license-server chart)

Reviewer: This should be part of the license server installation section

Where the license key can be any value and will later be used in connected operators and to manually fetch any statistic and license file must be a valid operator license that will be served to the connecting operators and should be mounted to the license server (when using chart can either be mounted from secret with license under `license.pem` key and possibly create said secret from `license.file.data` value).

## Tls

License server is able to self expose a https endpoint if needed by providing the relevant certifcate and private key or via `certManager` integration.

```yaml
# ./values.yaml
...

service:
  port: 433

tls:
  data:
    tls.key: |
      -----BEGIN PRIVATE KEY-----
      ...
      -----END PRIVATE KEY-----
    tls.crt: |
      -----BEGIN CERTIFICATE-----
      ...
      -----END CERTIFICATE-----
  # or
  certManager:
    enabled: true
```

*NOTE: If enabled certManager does create only certificates for internal dns names like `mirrord-operator-license-server` or `mirrord-operator-license-server.mirrord.svc` under a newly created `cert-manager.io/v1.Issuer` and does require `cert-manager.io` to be installed on the cluster*

> ## Ingress

Reviewer: Shouldn't this be around line 91?

### Helm Chart

To install the the license server with Helm, first add the MetalBear Helm repository:

```bash
helm repo add metalbear-co https://metalbear-co.github.io/charts
```

Now to setup the license-server as a standalone helm deployment copy the following yaml to a `values.yaml` file your local machine. 

```yaml
# ./values.yaml
createNamespace: true

service:
  type: ClusterIP

license:
  key: secret
  file:
    data:
      license.pem: |
        ----- ... 
        MIRRORD-LICENSE 
        ... -----
```

Then fill in the `license.key` and `license.file.data."license.pem"` with a license key value of your choosing and the contents of your operator license and/or customise the deployment to your choosing. (default `values.yaml` values can be found [here](https://raw.githubusercontent.com/metalbear-co/charts/main/mirrord-license-server/values.yaml))

*NOTE: The default value for `service.type` is `ClusterIP`. This service needs to be accessible to any mirrord operators you want to track, so you can also use `NodePort` or `LoadBalancer`*

> With `values.yaml` you can install the the license server on a kubernetes cluster.

Reviewer: ```suggestion
Next, install the license server on your cluster:
```

```bash
helm install metalbear-co/mirrord-operator-license-server -f ./values.yaml --generate-name --wait
```

> Once it's installed and running now you can connect the operator. The simplest way is to check that all the deployment pods are up.

Reviewer: ```suggestion
To make sure it's been installed successfully and is running:
```

```bash
kubectl get deployment -n mirrord mirrord-license-server
```

> If you have an ingress installed on the cluster please do expose the `mirrord-operator-license-server` service via http or https to the license server http/https endpoint (depends on `tls` and `service.port` values in license-server chart)

Reviewer: What do you mean "expose to the license server endpoint"? Shouldn't it be exposed to operators?

### Connecting Operators to the License Server

> Depending if installed the operator via helm or cli you can specify the following properties.

Reviewer: ```suggestion
Add the license server address to the operator configuration. If you've installed the operator using the mirrord CLI:
```

```bash
mirrord operator setup --accept-tos --license-key secret --license-server http://<license-server-addr> | kubectl apply -f -
```

> or if using with Helm chart

Reviewer: ```suggestion
Or if you installed it using Helm, first update your `values.yaml` file:
```

```yaml
# ./values.yaml
license:
  key: secret
  licenseServer: http://<license-server-addr>
```
*NOTE: The server value must contain the protocol and the prefix for any ingress that the the license server can be behind.*

> and then use the install or upgrade to set these parameters

Reviewer: ```suggestion
Then run:
```

---
title: "License Server"
description: "License Server"
date: 2025-04-07T00:00:00+00:00
lastmod: 2025-04-07T00:00:00+00:00
draft: false
images: []
linktitle: "License Server"
menu:
  docs:
    teams:
weight: 540
toc: true
tags: ["enterprise"]
---

The license server enables you to manage your organization’s seats without sending any data to mirrord’s servers. It can aggregate license metrics from multiple operators (useful if you’re running mirrord across multiple clusters) and provides visibility into seat usage across your organization.

## Basic Setup

The license server is installable via Helm. First, add the MetalBear Helm repository:

```bash
helm repo add metalbear-co https://metalbear-co.github.io/charts
```

Next, save the following yaml as `values.yaml` on your machine. 

```yaml
# ./values.yaml
createNamespace: true

service:
  type: ClusterIP

license:
  key: secret
  file:
    data:
      license.pem: |
        ----- ... 
        MIRRORD-LICENSE 
        ... -----
```

Fill in the license.key and license.pem fields according to the following guidelines:

> * License key - Can be any string of your choosing, we recommend you use a random characacters or a uuid for this value, because it will later be used in connected operators and to manually fetch usage statistics.

Reviewer: ```suggestion
* License key - Can be any string of your choosing. We recommend using a random characters or a UUID.
```

> * License file -  Must be a valid operator license that will be served to the connecting operators and should be mounted to the the license server (when using chart can either be mounted from secret with license under `license.pem` key and possibly create said secret from `license.file.data` value).

Reviewer: ```suggestion
* License file - Must be a valid operator license. This can also be a secret under the `license.pem` key.
```

---
title: "License Server"
description: "License Server"
date: 2025-04-07T00:00:00+00:00
lastmod: 2025-04-07T00:00:00+00:00
draft: false
images: []
linktitle: "License Server"
menu:
  docs:
    teams:
weight: 540
toc: true
tags: ["enterprise"]
---

The license server enables you to manage your organization’s seats without sending any data to mirrord’s servers. It can aggregate license metrics from multiple operators (useful if you’re running mirrord across multiple clusters) and provides visibility into seat usage across your organization.

## Basic Setup

The license server is installable via Helm. First, add the MetalBear Helm repository:

```bash
helm repo add metalbear-co https://metalbear-co.github.io/charts
```

Next, save the following yaml as `values.yaml` on your machine. 

```yaml
# ./values.yaml
createNamespace: true

service:
  type: ClusterIP

license:
  key: secret
  file:
    data:
      license.pem: |
        ----- ... 
        MIRRORD-LICENSE 
        ... -----
```

> Then fill in the `license.key` and `license.file.data."license.pem"` with a license key value of your choosing and the contents of your operator license and/or customise the deployment to your choosing. (all values.yaml configuration options can found [here](https://raw.githubusercontent.com/metalbear-co/charts/main/mirrord-license-server/values.yaml))

Reviewer: Change to the following:

"Fill in the `license.key` and `license.pem` fields according to the following guidelines:"
Then paste the content of the License section
"You can customize the license server deployment further - all configuration values can be found [here](...)."

> *NOTE: The default value for `service.type` is `ClusterIP`. This service needs to be accessible to any mirrord operators you want to track, so you can also use `NodePort` or `LoadBalancer`*

Reviewer: ```suggestion
*NOTE: The license server needs to be accessible to any mirrord operators you want to track. To that end, the default value for `service.type` is `ClusterIP`, but can be changed to `NodePort` or `LoadBalancer`, according to your requirements.*
```

Next, install the license server on your cluster:

```bash
helm install metalbear-co/mirrord-operator-license-server -f ./values.yaml --generate-name --wait
```

To make sure it's been installed successfully and is running:

```bash
kubectl get deployment -n mirrord mirrord-license-server
```

> If you have an ingress installed on the cluster please do expose the `mirrord-operator-license-server` service via http or https to the license server http/https endpoint (depends on `tls` and `service.port` values in license-server chart)

Reviewer: ```suggestion
If your operator(s) are running at on a different cluster, make sure the `mirrord-operator-license-server` service is exposed to them via ingress.
```

### Connecting Operators to the License Server

> If you installed operator using Helm, first update your operator `values.yaml` file: *(read more [here](/docs/overview/quick-start/#helm) for quickstart helm setup for operator)*

Reviewer: ```suggestion
First update your operator [`values.yaml`](/docs/overview/quick-start/#helm) for quickstart helm setup for operator) file:
```

```yaml
# ./values.yaml
license:
  key: secret
  licenseServer: http://<license-server-addr>
```

> *NOTE: The server value must contain the protocol and the prefix for any ingress that the the license server can be behind.*

Reviewer: I think we can drop everything until "Then run"
